src/main.rs

```diff
@@ -118,9 +133,17 @@ fn gumpi_async(opt: Opt, config: JobConfig) -> impl Future<Item = (), Error = fa
                 Either::B(future::ok(None))
             };
 
+            let upload_input = if let Some(input) = config.input.clone() {
+                let input_path = jobconfig_dir.clone().join(input.source);
```

Comment:
Is `clone` here necessary? Wouldn't

```rust
let input_path = jobconfig_dir.join(input.source);
```

work here as well since `join` returns a new, owned `PathBuf`?

Reply:
Good catch, thanks!